src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -31,3 +68,17 @@ const useTriggerProcessing = (props: { serialNumber: string; trigger: triggerTyp
   return { triggerProcessing: isProcessing, error: error || triggerError }
 }
 export default useTriggerProcessing
+
+function inferTriggerType(
+  triggerType: TriggerType | undefined,
+  serial: string | undefined,
+  reviewAssignmentId: number | undefined
+) {
+  return triggerType
+    ? triggerType
+    : serial
+    ? 'applicationTrigger'
+    : reviewAssignmentId
+    ? 'reviewAssignmentTrigger'
+    : 'reviewTrigger'
```

Comment:
I think you could be using the type here to return instead of strings

Reply:
Okay, I've added the `TriggerType` as the output of the function, so the returned values have to comply with that.

Is that what you mean, or were you meaning something else?

Reply:
Yes, that's fine. I was suggestion to also use the format  `TriggerType. applicationTrigger` for types, but not problem. Just using as the return will also flag for possible problem with types.